### DISCUSSION TITLE
Papers

### DISCUSSION THREAD

**Discussion by matanox** [online-ml/river#650] [Q&A] [answered]
What would be the best way to systematically learn the science behind this package? It might be a less obvious topic to find good reads about than e.g. the latest neural network architecture.

**Comment by MaxHalford** [comment, marked as answer]
Great question! As you say, online machine learning is not as popular as other subfields of machine learning, so there's much less papers and comprehensive reading.

Each River model has a `References` section where we list the relevant papers. That's the best we can do for now.

**Comment by matanox** [comment]
Thanks again for developing and maintaining this library and the two which are being merged/integrated into it! For some reason I can't click the "Mark as answer" new widget that Github place here, but this question can be closed obviously!

**Comment by matanox** [comment]
Just as one case in point though, is there any special paper or technique deeply involved in the incremental version of the [logistic regression implementation](https://riverml.xyz/latest/api/linear-model/LogisticRegression/)?

**Reply by MaxHalford** [reply]
For logistic regression, I would refer to [this part](https://scikit-learn.org/stable/modules/sgd.html) of scikit-learn's documentation. River's implementation produces the same results :)

**Comment by jacobmontiel** [comment]
Some resources that should provide some foundations:

- The “[Machine Learning for Data Streams with Practical Examples in MOA](https://moa.cms.waikato.ac.nz/book/)” textbook is a resource intended to help students and practitioners enter the field of machine learning and data mining for data streams.

- "[A survey on concept drift adaptation](https://dl.acm.org/doi/10.1145/2523813)". Although a little outdated, it is a great introduction to concept drift.

- "[Machine learning for streaming data: state of the art, challenges, and opportunities](https://dl.acm.org/doi/10.1145/3373464.3373470)"

This list is by no means complete, the field of data streams is continuously evolving (pun intended) and we expect it to keep growing.
